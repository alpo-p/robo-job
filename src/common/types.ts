import { Ionicons } from '@expo/vector-icons'

export type IoniconsIconNames = keyof typeof Ionicons.glyphMap

export interface IJobPostCard {
  id: string
  companyName: string
  jobTitle: string
  logoUrl: string
  pictureUrl: string
  location: string
  tags: string[]
  shortDescription: string
  longDescription: string
  deadline: string
  aboutTheCompany: string
  linkToWebsite: string
}
