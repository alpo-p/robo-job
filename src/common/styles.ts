export default {
  backgroundColor: '#000000',
  white: '#ffffff',
  darkBackgroundColor: '#222222',
  darkGrey: '#404040',
  mediumDarkGrey: '#d0d0d0',
  lightGrey: '#e0e0e0',

  successGreen: '#087c61',

  primaryColorBase: '#186db6',
  primaryColorDarkened10: '#00549D',
  primaryColorLightened10: '#3287D0',
  primaryColorDarkened20: '#003A83',
  primaryColorLightened20: '#4BA0E9',
  primaryColorDarkened30: '#00216A',

  backgroundWhite: 'rgb(242, 242, 242)',
  darkText: 'rgb(28, 28, 30)',
  whiteText: 'rgba(255, 255, 255, 0.87)',
  commonSize: 24,
  biggerCommonSize: 38,
}
