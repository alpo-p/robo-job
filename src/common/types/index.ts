export interface IJobPostCard {
  id: string
  companyName: string
  jobTitle: string
  logoUrl: string
  pictureUrl: string
  location: string
  tags: string[]
  shortDescription: string
  longDescription: string
  deadline: string
  aboutTheCompany: string
  linkToWebsite: string
}

export type MessageTypes = 'robo' | 'recruiter' | 'user'

export interface Message {
  typeOfMessage: MessageTypes
  text: string
}

export interface IndividualChat {
  jobPostId: string
  messages: Message[]
}

export type Chats = IndividualChat[]
