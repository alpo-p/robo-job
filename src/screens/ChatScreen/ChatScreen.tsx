import React from 'react'
import { Text } from 'react-native'
import { SafeContainer } from '../../sharedComponents/SafeContainer'

const ChatScreen = () => (
  <SafeContainer>
    <Text>Chat is rendered here</Text>
  </SafeContainer>
)

export default ChatScreen
