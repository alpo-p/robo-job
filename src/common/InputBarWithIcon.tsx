import { Ionicons } from '@expo/vector-icons'
import { useTheme } from '@react-navigation/native'
import React from 'react'
import { TextInput, useColorScheme, View } from 'react-native'
import styles from './styles'
import { IoniconsIconNames } from './types/utilTypes'

interface P {
  value: string
  // eslint-disable-next-line no-unused-vars
  setValue: (v: string) => void
  iconName: IoniconsIconNames
  placeholderText?: string
}

// TODO: refactor
export default ({
  value,
  setValue,
  iconName,
  placeholderText = 'Search',
}: P) => {
  const { colors } = useTheme()
  const scheme = useColorScheme()
  // TODO: refactor bgColor and placeholder color to themes
  const bgColor =
    scheme === 'dark' ? styles.lightBgDarkMode : styles.lightBgLightMode
  const placeholderColor =
    scheme === 'dark' ? styles.lighterWhiteText : styles.lighterDarkText
  return (
    <View
      style={{
        flexDirection: 'row',
        backgroundColor: bgColor,
        margin: 16,
        borderRadius: 8,
        height: 8 * 4,
      }}
    >
      <Ionicons
        style={{
          padding: 8,
        }}
        name={iconName}
        color={colors.primary}
        size={15}
      />
      <TextInput
        style={{
          color: colors.text,
          width: '70%',
        }}
        value={value}
        onChangeText={setValue}
        placeholder={placeholderText}
        placeholderTextColor={placeholderColor}
      />
    </View>
  )
}
