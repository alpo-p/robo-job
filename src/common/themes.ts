import {
  DarkTheme as DefaultDarkTheme,
  DefaultTheme,
} from '@react-navigation/native'
import styles from './styles'

export type ThemeColors = {
  primary: string
  background: string
  card: string
  text: string
  border: string
  notification: string
}

export const LightTheme = {
  ...DefaultTheme,
  colors: {
    ...DefaultTheme.colors,
    primary: styles.primaryColor,
    text: styles.darkText,
    icon: '#121212',
    background: styles.white,
  },
}

export const DarkTheme = {
  ...DefaultDarkTheme,
  colors: {
    ...DefaultDarkTheme.colors,
    primary: styles.primaryColor,
    text: styles.whiteText,
    card: '#121212',
    border: '#121212',
    icon: styles.whiteText,
  },
}
