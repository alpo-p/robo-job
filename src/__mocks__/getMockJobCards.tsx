import { IJobPostCard } from '../common/types'
import getMockJobPostCard from './getSingleMockJobCard'

export default function getMockJobCards(n = 10): IJobPostCard[] {
  const mockJobPostCards = [...Array(n).keys()].map(i => {
    const mockCard = getMockJobPostCard()
    mockCard.id = `${i}`
    mockCard.jobTitle = `Software developer #${i}`
    return mockCard
  })
  return mockJobPostCards
}
