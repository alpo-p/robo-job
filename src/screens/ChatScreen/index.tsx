import { RouteProp } from '@react-navigation/native'
import React from 'react'
import { SafeAreaView } from 'react-native-safe-area-context'
import { IJobPostCard } from '../../common/types'
import ChatTopBar from './components/ChatTopBar'
import MessagesContainer from './components/MessagesContainer'

interface P {
  route: RouteProp<{ params: { jobPost: IJobPostCard } }, 'params'>
}

export default ({
  route: {
    params: { jobPost },
  },
}: P) => (
  <SafeAreaView>
    <ChatTopBar jobPost={jobPost} />
    <MessagesContainer />
  </SafeAreaView>
)
