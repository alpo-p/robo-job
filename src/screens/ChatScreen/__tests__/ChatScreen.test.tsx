import { RouteProp } from '@react-navigation/native'
import React from 'react'
import { IJobPostCard } from '../../../common/types'
import create from '../../../testUtils/rendererCreateWithNavigation'
import getMockJobCards from '../../../__mocks__/getMockJobCards'
import ChatScreen from '../index'

describe('ChatScreen', () => {
  const prop = {
    params: { jobPost: getMockJobCards()[0] },
  } as unknown as RouteProp<{ params: { jobPost: IJobPostCard } }, 'params'>

  it('should match snapshot', () => {
    const tree = create(<ChatScreen route={prop} />)
    expect(tree).toMatchSnapshot()
  })
})
