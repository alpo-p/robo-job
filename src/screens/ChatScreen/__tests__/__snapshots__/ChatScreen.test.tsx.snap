// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatScreen should match snapshot 1`] = `
<RNCSafeAreaView>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#000000",
        "flexDirection": "row",
        "height": 64,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": undefined,
          "marginLeft": 8,
          "opacity": 1,
          "position": "relative",
          "top": undefined,
          "zIndex": 1,
        }
      }
    >
      <Text />
    </View>
    <View
      style={
        Object {
          "marginRight": 8,
        }
      }
    />
    <Image
      source={
        Object {
          "uri": "https://ih1.redbubble.net/image.659413411.3655/st,small,507x507-pad,600x600,f8f8f8.u1.jpg",
        }
      }
      style={
        Object {
          "borderRadius": 24,
          "height": 48,
          "width": 48,
        }
      }
    />
    <View
      style={
        Object {
          "marginLeft": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Damascus",
            "fontSize": 14,
            "textTransform": "none",
          }
        }
      >
        E Corp
      </Text>
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "DamascusLight",
            "fontSize": 14,
            "textTransform": "none",
          }
        }
      >
        Digital Marketing Manager
      </Text>
    </View>
    <View
      style={
        Object {
          "justifyContent": "center",
          "marginLeft": "auto",
          "marginRight": 8,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "height": "92%",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RCTScrollView
      onContentSizeChange={[Function]}
    >
      <View />
    </RCTScrollView>
    <View
      style={
        Object {
          "backgroundColor": "#000000",
          "borderColor": "#000000",
          "borderRadius": 16,
          "borderWidth": 1,
          "flexDirection": "row",
          "margin": 16,
          "marginBottom": 16,
          "padding": 8,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        multiline={true}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Write your answer here..."
        placeholderTextColor="rgba(28, 28, 30, 0.4)"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#000000",
            "color": "#000000",
            "flex": 5,
            "padding": 4,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "marginLeft": 8,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000000",
              "fontFamily": "DamascusBold",
              "fontSize": 16,
              "textTransform": "none",
            }
          }
        >
          Send
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
